Geometry2d/Point.hpp

```diff
@@ -157,6 +163,24 @@ class Point {
         return x() != other.x() || y() != other.y();
     }
 
+    double& operator[](int i) {
+        if (0 == i) {
+            return _x;
+        } else if (1 == i) {
+            return _y;
+        } else {
+            throw std::out_of_range("Out of range index for Geometry2d::Point");
+        }
+    }
+
+    /**
+     * Hash function for Geometry2d::Point
+     */
+    static size_t hash(Point pt) {
+        return std::hash<double>{}(pt.x()*233 + pt.y());
```

Comment:
Oooh. You should use hashcombine. it looks fun.

http://www.boost.org/doc/libs/1_35_0/doc/html/hash/combine.html